### PR TITLE
Revamp close button component

### LINKED ISSUE #27163: Restyle close icon's focus style
This button shouldn't be styled that way when focused (clicked) :(

image

### LINKED ISSUE #27294: Rename close icon class to btn-close
Imo:
1. `close` name is very generic, but adds a lot of styling and interferes with any use of such word for other purposes. Example: For buttons with icons I often use `btn btn-secondary <icon-name>`, If my icon is named `close` the button will get a lot of unwanted styles. 
2. Close icon's use-case is generally a button, so why not `btn btn-close`?

### LINKED ISSUE #28787: [feature] toast-header .close
Modal has padding and margin for the close button (.close)
```scss
.modal-header {
  display: flex;
  align-items: flex-start; // so the close btn always stays on the upper right corner
  justify-content: space-between; // Put modal header elements (title and dismiss) on opposite ends
  padding: $modal-header-padding;
  border-bottom: $modal-header-border-width solid $modal-header-border-color;
  @include border-top-radius($modal-content-border-radius);

  .close {
    padding: $modal-header-padding;
    // auto on the left force icon to the right even when there is no .modal-title
    margin: (-$modal-header-padding-y) (-$modal-header-padding-x) (-$modal-header-padding-y) auto;
  }
}
```

I suggest we add the same thing to toasts

```scss
.toast-header {
  display: flex;
  align-items: center;
  padding: $toast-padding-y $toast-padding-x;
  color: $toast-header-color;
  background-color: $toast-header-background-color;
  background-clip: padding-box;
  border-bottom: $toast-border-width solid $toast-header-border-color;

// add this ----->
  .close {
    padding: $toast-padding-y $toast-padding-x;
    margin-right: auto;
  }
}
```

Maybe add a `$toast-header-padding`, `$toast-header-padding-x`, `$toast-header-padding-y` variables and margin?

The main thing is the `margin-right` or something that does its job.

### LINKED ISSUE #30917: Bootstrap 4.5 or browser(s) bug ? modal close button with frame ?
go to Bootstrap page :

https://getbootstrap.com/docs/4.5/components/modal/
https://getbootstrap.com/docs/4.0/components/modal/

Then click on sample modal dismiss button (right upper corner). See the frame around ? 
**Wasn't there two days ago.**

it is in 
Chome 83.0.4103.61 Win7 64, 
new Edge based on chromium engine, 
latest Firefox 75.

When set css element from
pointer-events:auto
to
pointer-events: none

frame gone, but loose some visual behaviour.

regards.

@XhmikosR, next time check before You close an issue. I Know what ama writing. This problem is new.

### PR DESCRIPTION
**Pull request by mdo**
This PR redesigns our close buttons to make them more consistent across browsers and devices, as well as making it slightly easier to style them overall. With a little CSS `background-image` magic, we're sizing the &times; SVG to the `content-box`. This allows us to give use `padding` and rounded corners that carry the styling through with no impact to the &times; icon.

- Renamed `.close` to `.btn-close` for more specific class name
- Replaced inline `&times;` sign with embedded SVG in our CSS
- Improved focus styles with custom `box-shadow` "outline" like our buttons
- Increased focus shadow width of all buttons, including the new close button, from `.2em` to `.25em` for a more pixel perfect rendering
- Adds `.btn-close-white` variant for darker backgrounds.

Here's a preview of the default and disabled styles, as well as a focused and disabled style.

Screen Shot 2020-09-13 at 7 12 22 PM

Screen Shot 2020-09-13 at 7 12 29 PM

Screen Shot 2020-09-13 at 7 47 56 PM

---

Closes #30927 and closes #31478.

Fixes #30917 (focus), fixes #28787 (default styling in toasts), fixes #27294 (rename). and fixes #27163 (focus).

Preview: <https://deploy-preview-31641--twbs-bootstrap.netlify.app/docs/5.0/components/close-button/>

### CODE REVIEW

**Comment by mdo** [comment]
Without adding another variable, I think the only downside to the `invert()` on the white variant is the focus state gets inverted, too. I assumed this would only apply to the `background-image`. Might need to research more, or fallback to redeclaring the variable.

**Comment by XhmikosR** [comment]
@mdo: How does this look on Legacy Edge? As long as it sort of falls back to something and is acceptable, it's not a big deal if it's not supported.

https://caniuse.com/css-filters

**Comment by mdo** [comment]
Looks good on Edge 18.

Screen Shot 2020-09-18 at 3 26 32 PM

**Comment by XhmikosR** [comment]
Maybe the `btn-close-white` focus border color could be a little whiter?

**Comment by mdo** [comment]
> Maybe the `btn-close-white` focus border color could be a little whiter?

Added `brightness(200%)` to tackle that.

**Review by XhmikosR — APPROVED** [review]
LGTM, maybe wait for @twbs/css-review for another approval

File: scss/_alert.scss
Comment (ffoodd) [inline scss/_alert.scss:30]: `3.5` and `1.25` above look  a bit random here. Does it match anything? Shouldn't they be handled as variables?

File: scss/_close.scss
Comment (ffoodd) [inline scss/_close.scss:40]: Could it be useful to set this value as a variable, too?

**Review by ffoodd — APPROVED** [review]
Except a few hardcoded values, looks good to me 👌

**Comment by mdo** [comment]
Nice, great comments @ffoodd! Updated the alert CSS to include two new variables. Also changed the `3.5` to `3` (previous value was too much) and explained why we use that value.

**Pull request merged**
